OpenLIFUSonicationControl/OpenLIFUSonicationControl.py

```diff
@@ -567,3 +789,48 @@ def create_openlifu_run(self, run_parameters: Dict) -> SlicerOpenLIFURun:
         slicer.util.getModuleLogic('OpenLIFUData').set_run(run)
         
         return run
+
+    def reinitialize_lifu_interface(self, test_mode=False):
+        """ Reinitializes the LIFUInterface and switches out the monitoring task
+        with the test mode version """
+
+        # Note: in test_mode, the true start_monitoring() was never called
+
+        self.cur_lifu_interface.stop_monitoring()
+
+        # Cancel current task if it's running
+
+        if self.current_monitoring_task and not self.current_monitoring_task.done():
+            self.current_monitoring_task.cancel()
+            try:
+                self.monitoring_loop.run_until_complete(self.current_monitoring_task)  # Ensure it's fully canceled
+            except asyncio.CancelledError:
+                pass
```

Comment:
Having to do this makes me feel like we are doing something wrong, but maybe just worth revisiting in the future after clarifying our understanding of asyncio. It's fine for now, I don't think tying together two event loops is going to be trivial in any world!